### ISSUE TITLE
Can Not Match the Same mAP retraining Group Free 3D

### ISSUE THREAD

**Issue by stanleyshly** [open-mmlab/mmdetection3d#1107]
I am currently trying to retrain Group Free 3D, more specifically, the L6, O256 version of the model. However, since I only have one GPU with only 3GB of vram, I can only fit one batch at time. 
As a result, I decided to try the Gradeint Cumulation feature. I have made only two changes from the default config file for the L6 O256 Vanilla PointNet++ Backbone version.
`optimizer_config = dict(grad_clip=dict(max_norm=0.1, norm_type=2), type='GradientCumulativeOptimizerHook', cumulative_iters=32)`
and
` samples_per_gpu=1,`
However, after training, my mAP and even mAR are a quite a bit far from expected. However, based on the readme in the config file, I am expected around a 65 mAP@0.25 and a 47 mAP@0.5, while my results are closer to 50 mAP@0.25 and a 29 mAP@0.5. This is a quite a difference, can someone help me debug this issue, or is the Gradient Cumulation feature just broken? I tried scaling my LR by sqrt{new batch size/old batch size}, without gradient cumulation, but I think the model converge early and got stuck in a local minima.
My retrained eval results:
```
+----------------+---------+---------+---------+---------+
| classes        | AP_0.25 | AR_0.25 | AP_0.50 | AR_0.50 |
+----------------+---------+---------+---------+---------+
| cabinet        | 0.3393  | 0.7016  | 0.0960  | 0.3656  |
| bed            | 0.8411  | 0.9506  | 0.8085  | 0.8765  |
| chair          | 0.8342  | 0.9115  | 0.5917  | 0.6988  |
| sofa           | 0.8027  | 0.9588  | 0.5725  | 0.7216  |
| table          | 0.4992  | 0.7857  | 0.3492  | 0.5743  |
| door           | 0.4135  | 0.6916  | 0.1558  | 0.3448  |
| window         | 0.3151  | 0.5496  | 0.0998  | 0.1915  |
| bookshelf      | 0.2724  | 0.8052  | 0.1933  | 0.4675  |
| picture        | 0.0559  | 0.1712  | 0.0079  | 0.0360  |
| counter        | 0.3238  | 0.6923  | 0.0406  | 0.2308  |
| desk           | 0.5663  | 0.9370  | 0.3461  | 0.7402  |
| curtain        | 0.3951  | 0.7612  | 0.1201  | 0.2687  |
| refrigerator   | 0.3615  | 0.8596  | 0.2848  | 0.6491  |
| showercurtrain | 0.4761  | 0.7857  | 0.1423  | 0.2857  |
| toilet         | 0.8584  | 0.9828  | 0.6025  | 0.7241  |
| sink           | 0.5475  | 0.6939  | 0.2342  | 0.3776  |
| bathtub        | 0.7571  | 0.9355  | 0.5349  | 0.6452  |
| garbagebin     | 0.3773  | 0.6774  | 0.1923  | 0.4302  |
+----------------+---------+---------+---------+---------+
| Overall        | 0.5020  | 0.7695  | 0.2985  | 0.4793  |
+----------------+---------+---------+---------+---------+
```

**Comment by Tai-Wang** [comment]
Do you mean that you only use batch size 1 with one GPU to run the experiment? Actually, you will encounter a lot of problems with such a little batch, such as tuning the learning rate is really tricky and maybe BN can be a bit unstable for this case. I would recommend you try more choices on these settings.

**Comment by stanleyshly** [comment]
> Do you mean that you only use batch size 1 with one GPU to run the experiment? Actually, you will encounter a lot of problems with such a little batch, such as tuning the learning rate is really tricky and maybe BN can be a bit unstable for this case. I would recommend you try more choices on these settings.

Yes, but I used Gradient Cumulation to "simulate" a batch size of 32, and kept the normal settings as a result. Even with gradient cumulation, would you recommend increasing the LR or decreasing the LR?

**Comment by Tai-Wang** [comment]
We can not guarantee that works for our supported models because there is no attempt before. We will try to reproduce your case and give some feedback here.

**Comment by stanleyshly** [comment]
Thank you for your help. I had guessed that gradient accumulation would be able to basically work with the same LR as with normal settings, but I guess the BN instability has made things worse than normal, by a significant amount.

**Comment by Tai-Wang** [comment]
Temporarily close this issue. If you have any progress or results required discussion, please feel free to reopen it.
